### PR TITLE
chore(release): add beta2 charts for workflow and workflow-e2e

### PR DESCRIPTION
**Pull request by arschles**
This PR adds charts for the beta2 release of the workflow and workflow-e2e charts

TODO:
- [x] Execute tests on the workflow-beta2 chart.
  - Create an issue for building a jenkins job that does this automatically
- [x] Tag & push beta2 docker images
- [x] Change `workflow-beta2/tpl/generate_params.toml` to make all components point to the `deis` org and have `v2.0.0-beta2` docker tags
- [x] Update changelog
- [x] Tag and push git repos

### CODE REVIEW

**Comment by sgoings** [comment]
@arschles this job https://ci.deis.io/job/workflow-test-beta2 tests this PR:
1. checks out your arschles/deis-charts repo with branch = release-beta2
2. sets the following environment variables
   
   ```
   WORKFLOW_CHART=workflow-beta2
   WORKFLOW_E2E_CHART=workflow-beta2-e2e
   ```
3. doesn't commit/push back to your branch

**Comment by sgoings** [comment]
The Jenkins initiated test (https://ci.deis.io/job/workflow-test-beta2/2/console) failed because the referenced image for minio in the kubernetes manifest: `quay.io/deisci/minio:git-e7f7ca1` does not exist: https://quay.io/repository/deisci/minio?tab=tags.

File: workflow-beta2/manifests/deis-workflow-manager-rc.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    app: deis-workflow-manager
+  template:
+    metadata:
+      labels:
+        app: deis-workflow-manager
+    spec:
+      serviceAccount: deis-workflow-manager
+      containers:
+      - name: deis-workflow-manager
+        image: quay.io/deisci/workflow-manager:git-9ba6db7
```
Comment (kmala) [inline workflow-beta2/manifests/deis-workflow-manager-rc.yaml]: shouldn't image should be generated using helm generate?

File: workflow-beta2/tpl/deis-builder-rc.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+              name: healthsrv
+          env:
+            - name: "HEALTH_SERVER_PORT"
+              value: "8092"
+            - name: "EXTERNAL_PORT"
+              value: "2223"
+            - name: BUILDER_STORAGE
+              value: {{default "minio" .storage}}
+            - name: "SLUGBUILDER_IMAGE_NAME"
+              value: "quay.io/deisci/slugbuilder:git-a713e13"
```
Comment (kmala) [inline workflow-beta2/tpl/deis-builder-rc.yaml]: shouldn't image should be generated using helm generate

**Comment by arschles** [comment]
@kmala you left some comments but I don't see them in here. anyway, relevant code has been updated

File: workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#helm:generate helm tpl -d $HELM_GENERATE_DIR/tpl/generate_params.toml -o $HELM_GENERATE_DIR/manifest/deis-logger-fluentd-daemon.yaml $HELM_GENERATE_DIR/tpl/deis-logger-fluentd-daemon.yaml
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: deis-logger-fluentd
+  namespace: deis
+  labels:
+    heritage: deis
+spec:
```
Comment (jchauncey) [inline workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml:9]: This section needs to look like this 

``` yaml
spec:
  selector:
    matchLabels:
      app: deis-logger-fluentd
      heritage: deis
  template:
    metadata:
      name: deis-logger-fluentd
      labels:
        heritage: deis
        app: deis-logger-fluentd
```

Comment (arschles) [inline workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml:9]: @jchauncey that's the entire `spec`? This is copied from [the one in `workflow-dev`](https://github.com/deis/charts/blob/master/workflow-dev/manifests/deis-logger-fluentd-daemon.yaml#L19-L44)

Comment (jchauncey) [inline workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml:9]: not its just the part to the next `spec:` here is the entire thing you need - 

```
spec:
  selector:
    matchLabels:
      app: deis-logger-fluentd
      heritage: deis
  template:
    metadata:
      name: deis-logger-fluentd
      labels:
        heritage: deis
        app: deis-logger-fluentd
    spec:
      serviceAccount: deis-logger-fluentd
      containers:
      - name: deis-logger-fluentd
        image: quay.io/deisci/fluentd:canary
        imagePullPolicy: Always
        securityContext:
          privileged: true
        volumeMounts:
        - name: varlog
          mountPath: /var/log
        - name: varlibdockercontainers
          mountPath: /var/lib/docker/containers
          readOnly: true
        env:
        - name: SYSLOG_HOST
          value: $(DEIS_LOGGER_SERVICE_HOST)
        - name: SYSLOG_PORT
          value: $(DEIS_LOGGER_SERVICE_PORT_TRANSPORT)
      volumes:
      - name: varlog
        hostPath:
          path: /var/log
      - name: varlibdockercontainers
        hostPath:
          path: /var/lib/docker/containers
```

**Comment by slack** [comment]
The diff in https://github.com/deis/charts/pull/222/files may help clarify?

File: workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml (hunk 3 above)
Comment (arschles) [inline workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml:9]: @jchauncey thanks. missing pieces added in https://github.com/deis/charts/pull/224/commits/ede8259117f7362f0828b79ba960732d05f612fb

**Comment by bacongobbler** [comment]
For anyone else trying to check this out, the remote branch is in arschles/deis-charts. arschles/charts redirects to arschles/helm-charts which does not have this branch. :)

**Comment by arschles** [comment]
Thanks @bacongobbler - the branch URL is https://github.com/arschles/deis-charts/tree/release-beta2 to be clear

**Comment by kmala** [comment]
@bacongobbler the branch is https://github.com/arschles/deis-charts

**Comment by bacongobbler** [comment]
I can't seem to get it to generate:

```
><> helm generate workflow-beta2
[ERROR] Failed to open /home/bacongobbler/.helm/workspace/charts/workflow-beta2/manifest/deis-logger-fluentd-daemon.yaml: open /home/bacongobbler/.helm/workspace/charts/workflow-beta2/manifest/deis-logger-fluentd-daemon.yaml: no such file or directory
[ERROR] Failed to complete generation: failed to execute helm tpl -d /home/bacongobbler/.helm/workspace/charts/workflow-beta2/tpl/generate_params.toml -o /home/bacongobbler/.helm/workspace/charts/workflow-beta2/manifest/deis-logger-fluentd-daemon.yaml /home/bacongobbler/.helm/workspace/charts/workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml (/home/bacongobbler/.helm/workspace/charts/workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml): exit status 1
```

Notice that the path references `manifest`, not `manifests`.

File: workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+#helm:generate helm tpl -d $HELM_GENERATE_DIR/tpl/generate_params.toml -o $HELM_GENERATE_DIR/manifest/deis-logger-fluentd-daemon.yaml $HELM_GENERATE_DIR/tpl/deis-logger-fluentd-daemon.yaml
```
Comment (bacongobbler) [inline workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml]: change `manifest` to `manifests/deis-logger-fluentd-...`

Comment (arschles) [inline workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml]: good catch. fixed inhttps://github.com/deis/charts/pull/224/commits/88090d645a43c51f0e60c7acc72d9a5e275a1676 and restarting the [failed jenkins job](https://ci.deis.io/job/workflow-test-beta2/8/console)

**Comment by bacongobbler** [comment]
There actually seems to be a few problem spaces:

```
><> git grep manifest/
workflow-beta2/tpl/deis-logger-fluentd-daemon.yaml:#helm:generate helm tpl -d $HELM_GENERATE_DIR/tpl/generate_params.toml -o $HELM_GENERATE_DIR/manifest/deis-logger-fluentd-daemon.yaml $HELM_GENERATE_DIR/tpl/deis-logger-fluentd-daemon.yaml
workflow-beta2/tpl/deis-logger-rc.yaml:#helm:generate helm tpl -d $HELM_GENERATE_DIR/tpl/generate_params.toml -o $HELM_GENERATE_DIR/manifest/deis-logger-rc.yaml $HELM_GENERATE_DIR/tpl/deis-logger-rc.yaml
workflow-beta2/tpl/deis-minio-rc.yaml:#helm:generate helm tpl -d $HELM_GENERATE_DIR/tpl/generate_params.toml -o $HELM_GENERATE_DIR/manifest/deis-minio-rc.yaml $HELM_GENERATE_DIR/tpl/deis-minio-rc.yaml
workflow-beta2/tpl/deis-router-rc.yaml:#helm:generate helm tpl -d $HELM_GENERATE_DIR/tpl/generate_params.toml -o $HELM_GENERATE_DIR/manifest/deis-router-rc.yaml $HELM_GENERATE_DIR/tpl/deis-router-rc.yaml
```

**Comment by arschles** [comment]
@bacongobbler good catch. I've fixed the typos in https://github.com/deis/charts/pull/224/commits/88090d645a43c51f0e60c7acc72d9a5e275a1676 and https://github.com/deis/charts/pull/224/commits/ad67996977cac1b2d70f70298f21447cd748e0ae, and restarted the [jenkins job](https://ci.deis.io/job/workflow-test-beta2)

**Comment by bacongobbler** [comment]
After fixing those, I hit this error:

```
><> helm generate workflow-beta2
[ERROR] Failed: template: helmTpl:21: unexpected bad character U+002D '-' in command
[ERROR] Failed to complete generation: failed to execute helm tpl -d /home/bacongobbler/.helm/workspace/charts/workflow-beta2/tpl/generate_params.toml -o /home/bacongobbler/.helm/workspace/charts/workflow-beta2/manifests/deis-workflow-manager-rc.yaml /home/bacongobbler/.helm/workspace/charts/workflow-beta2/tpl/deis-workflow-manager-rc.yaml (/home/bacongobbler/.helm/workspace/charts/workflow-beta2/tpl/deis-workflow-manager-rc.yaml): exit status 1
```

Looking into it now...

**Comment by arschles** [comment]
@bacongobbler I think I see the problem. the template variable is `.workflow-manager`, which is illegal. fix incoming

**Comment by bacongobbler** [comment]
Yep, that's the one! Changing it to `workflowManager` did the trick. I can finally generate now.

**Comment by arschles** [comment]
@bacongobbler :+1: - fixed in https://github.com/deis/charts/pull/224/commits/42d156524ed5b4fc65d693309329af21065b27c1

**Comment by smothiki** [comment]
No LGTM fluentd daemon set is missing resource constraints
ping @jchauncey

**Comment by smothiki** [comment]
refer https://github.com/deis/logger/blob/master/manifests/deis-logger-fluentd-daemon.yaml

**Comment by jchauncey** [comment]
i dont think that will help here. im not sure what you are seeing is because of a lack of constraint. i think those pods are killing the daemon fro the same reason that fluentd was killing the workflow-e2e test pods yesterday

**Comment by smothiki** [comment]
refer https://github.com/deis/logger/issues/61
lets address this in next release LGTM

**Pull request merged**